config.go

```diff
@@ -29,15 +29,16 @@ import (
 // Config contains configuration for a client.
 // Zero value of Config is the default configuration.
 type Config struct {
-	lifecycleListeners  map[types.UUID]LifecycleStateChangeHandler
-	membershipListeners map[types.UUID]cluster.MembershipStateChangeHandler
-	ClientName          string                 `json:",omitempty"`
-	Logger              logger.Config          `json:",omitempty"`
-	Failover            cluster.FailoverConfig `json:",omitempty"`
-	Labels              []string               `json:",omitempty"`
-	Serialization       serialization.Config   `json:",omitempty"`
-	Cluster             cluster.Config         `json:",omitempty"`
-	Stats               StatsConfig            `json:",omitempty"`
+	lifecycleListeners      map[types.UUID]LifecycleStateChangeHandler
+	membershipListeners     map[types.UUID]cluster.MembershipStateChangeHandler
+	FlakeIDGeneratorConfigs map[string]FlakeIDGeneratorConfig `json:",omitempty"`
```

Comment:
I considered not exporting this field and exposing some methods like `AddFlakeIDGeneratorConfig` or so. However, that way JSON support wouldn’t cover id configs. And once the field is exported, `Add*` alike methods would not be reasonable to expose. Hence it's now users' responsibility to take care of nil map assignments - although I think it's not the best way in terms of user experience. WDYT @yuce?

Reply:
addressed in 8a750b7